### DISCUSSION TITLE
Prisma Middleware with NestJs

### DISCUSSION THREAD

**Discussion by karatayburakk** [prisma/prisma#13778] [Q&A]
I'm trying to add middleware to my Nestjs project. I don't know the exact location to put this middleware. I tried to add in the PrismaService class constructor but it doesn't work correctly.

```ts
export class PrismaService extends PrismaClient implements OnModuleInit {
  constructor(config: ConfigService) {
    super({
      datasources: {
        db: {
          url: config.get('DATABASE_URL'),
        },
      },
      rejectOnNotFound: {
        findUnique: (err) => new NotFoundException(err.message),
        findFirst: (err) => new NotFoundException(err.message),
      },
    });

    this.$use(async (params, next) => {
      console.log(1);
      const result = await next(params);
      console.log(2);
      return result;
    });
  }

  async onModuleInit() {
    await this.$connect();
  }

  async enableShutdownHooks(app: INestApplication) {
    this.$on('beforeExit', async () => {
      await app.close();
    });
  }
}
```

After that, I request a handler to create something but the output is wrong, something like that:
1
2
1
2
1

Where is the exact location to put this middleware in NestJs?

**Comment by austincrim** [comment]
Hey @karatayburakk 👋,

Could you give me some more detail about where you use this service in your code and how you are making requests?

**Reply by vladi-stevanovic** [reply]
👋 Hello @karatayburakk  I'm going through pending GitHub Discussions to check if you need any help. Is this issue still open?
Please let us know if we can investigate it further or if you've already solved it, we'd love to know how!

It would be super useful for other users facing this problem! 🙌

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
